File: skellig-test-step-processing/src/main/java/org/skellig/teststep/processing/converter/DateTimeValueConverter.java
package org.skellig.teststep.processing.converter;

import org.apache.commons.lang3.StringUtils;

import java.time.DateTimeException;
import java.time.LocalDateTime;
import java.time.ZoneId;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

class DateTimeValueConverter implements TestStepValueConverter {

    private static final Pattern NOW_REGEX = Pattern.compile("now\\(([\\w]*)\\)");

    @Override
    public Object convert(String value) {
        Matcher matcher = NOW_REGEX.matcher(value);
        if (matcher.find()) {
            String timezone = matcher.group(1);
            return getLocalDateTime(timezone);
        }
        return value;
    }

    private Object getLocalDateTime(String timezone) {
        LocalDateTime localDateTime;
        if (StringUtils.isEmpty(timezone)) {
            localDateTime = LocalDateTime.now();
        } else {
            try {
                localDateTime = LocalDateTime.now(ZoneId.of(timezone));
            } catch (DateTimeException ex) {
                localDateTime = LocalDateTime.now();
            }
        }
        return localDateTime;
    }

}

File: skellig-test-step-processing/src/main/kotlin/org/skellig/teststep/processing/converter/DateTimeValueConverter.kt
package org.skellig.teststep.processing.converter

import org.apache.commons.lang3.StringUtils
import java.time.DateTimeException
import java.time.LocalDateTime
import java.time.ZoneId
import java.util.regex.Pattern

class DateTimeValueConverter : TestStepValueConverter {

    companion object {
        private val NOW_REGEX = Pattern.compile("now\\(([\\w]*)\\)")
    }

    override fun convert(value: String?): Any? {
        val matcher = NOW_REGEX.matcher(value ?: "")
        if (matcher.find()) {
            val timezone = matcher.group(1)
            return getLocalDateTime(timezone)
        }
        return value
    }

    private fun getLocalDateTime(timezone: String): Any? {
        return if (StringUtils.isEmpty(timezone)) {
            LocalDateTime.now()
        } else {
            try {
                LocalDateTime.now(ZoneId.of(timezone))
            } catch (ex: DateTimeException) {
                LocalDateTime.now()
            }
        }
    }

}